src/SignalR/clients/java/signalr/core/src/main/java/com/microsoft/signalr/HubConnection.java

```diff
@@ -264,7 +264,16 @@ public Completable start() {
                 Transport transport = customTransport;
                 if (transport == null) {
                     Single<String> tokenProvider = negotiateResponse.getAccessToken() != null ? Single.just(negotiateResponse.getAccessToken()) : accessTokenProvider;
-                    switch (negotiateResponse.getChosenTransport()) {
+                    TransportEnum chosenTransport;
+                    if (this.skipNegotiate) {
+                        if (this.transportEnum != TransportEnum.WEBSOCKETS) {
+                            throw new RuntimeException("Negotiation can only be skipped when using the WebSocket transport directly.");
```

Comment:
I think we could make it a bit more clear in the doc comments that the transport must explicitly be set to `{@link TransportEnum.WEBSOCKETS}` rather than it being something that happens automatically when you call `.shouldSkipNegotiate(true)`.

https://github.com/dotnet/aspnetcore/blob/c925f99cddac0df90ed0bc4a07ecda6b054a0b02/src/SignalR/clients/java/signalr/core/src/main/java/com/microsoft/signalr/HttpHubConnectionBuilder.java#L70-L77

Reply:
Let this ruminate over the weekend, and I'm going to remove this check from the 5.0 change and add it to the main branch.
This would break folks who didn't set a transport and it would use WebSockets first by default but now would throw.